docs/reference/mapping/types/dense-vector.asciidoc

```diff
@@ -134,9 +148,16 @@ distance) between the vectors. The document `_score` is computed as
 
 `dot_product`:::
 Computes the dot product of two vectors. This option provides an optimized way
-to perform cosine similarity. In order to use it, all vectors must be of unit
-length, including both document and query vectors. The document `_score` is
-computed as `(1 + dot_product(query, vector)) / 2`.
+to perform cosine similarity. The constraints and computed score are defined
+by `element_type`.
++
+When `element_type` is `float`, all vectors must be unit length, including both
+document and query vectors. The document `_score` is computed as
+`(1 + dot_product(query, vector)) / 2`.
++
+When `element_type` is `byte`, all vectors must have the same
+length including both document and query vectors or results will be inaccurate.
+The document `_score` is computed as `0.5 + dot_product(query, vector) / 32768`.
```

Comment:
Tiny comment, I think it'd be clearer if we wrote `(1 + dot_product(query, vector)) / 32768)` to match the equation above.

Reply:
I'm glad you had me double check as I actually missed the dimensions as part of the equation for this. Let me fix this. Thank you!

Reply:
I updated this to `0.5 + (dot_product(query, vector) / (32768 * dims))` . With dims I think it would be trickier to multiple through on the `0.5` now, so I would favor leaving it like this.